nixos/tests/systemd-initrd.nix

```diff
@@ -0,0 +1,11 @@
+import ./make-test-python.nix ({ pkgs, ... }: {
+  name = "systemd-initrd";
+
+  machine = { pkgs, ... }: {
+    boot.initrd.systemd.enable = true;
+  };
+
+  testScript = ''
+    machine.wait_for_unit("basic.target")
```

Comment:
Maybe also test `autoResize`/`autoFormat`?

Reply:
`autoFormat` is implicitly tested because the machines in a nixos test use it to format their root disks. But an `autoResize` test would be good as well.

Reply:
I've added such a test